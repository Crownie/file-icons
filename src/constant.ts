export const FILE_TYPE_MAP = {
  folder: ['folder'],
  folderOpen: ['folder-open'],
  ai: ['ai', 'eps'],
  app: ['app'],
  axure: ['rp'],
  book: [
    'mobi',
    'oeb',
    'lit',
    'xeb',
    'ebx',
    'rb',
    'pdb',
    'epub',
    'azw3',
    'hlp',
    'chm',
    'wdl',
    'ceb',
    'abm',
    'pdg',
    'caj',
  ],
  css: ['css', 'less', 'sass'],
  dmg: ['dmg'],
  excel: [
    'csv',
    'fods',
    'ods',
    'ots',
    'xls',
    'xlsm',
    'xlsx',
    'xlt',
    'xltm',
    'xltx',
  ],
  exe: ['exe'],
  html: ['htm', 'html', 'mht'],
  img: [
    'png',
    'bmp',
    'jpg',
    'jpeg',
    'gif',
    'webp',
    'tga',
    'exif',
    'fpx',
    'svg',
    'hdri',
    'raw',
  ],
  java: ['jar', 'java'],
  js: ['js'],
  json: ['json'],
  keynote: ['key'],
  md: ['md', 'markdown'],
  music: [
    'mp3',
    'cda',
    'wav',
    'wma',
    'ra',
    'mida',
    'ogg',
    'ape',
    'flac',
    'aac',
  ],
  numbers: ['numbers'],
  pages: ['pages'],
  pdf: ['pdf'],
  php: ['php'],
  pkg: ['pkg'],
  ppt: [
    'fodp',
    'odp',
    'otp',
    'pot',
    'potm',
    'potx',
    'pps',
    'ppsm',
    'ppsx',
    'ppt',
    'pptm',
    'pptx',
  ],
  psd: ['psd'],
  python: ['python'],
  sh: ['sh'],
  sketch: ['sketch'],
  sql: ['sql'],
  text: ['text', 'txt'],
  video: ['mp4', 'avi', 'mov', 'rmvb', 'rm', 'flv', 'mpeg', 'wmv', 'mkv'],
  vue: ['vue'],
  word: [
    'doc',
    'docm',
    'docx',
    'dot',
    'dotm',
    'dotx',
    'epub',
    'fodt',
    'odt',
    'ott',
    'rtf',
  ],
  xmind: ['xmind'],
  zip: [
    'zip',
    'rar',
    'tar',
    'gz',
    'gzip',
    'uue',
    'bz2',
    'iso',
    '7z',
    'z',
    'ace',
    'lzh',
    'arj',
    'cab',
  ],
};
