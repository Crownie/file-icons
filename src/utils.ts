import {FILE_TYPE_MAP} from './constant';
import kebabCase from 'lodash.kebabcase';
export const getIconFileName = function getIconType(ext: string): string {
  let fileType = 'common';
  Object.keys(FILE_TYPE_MAP).map(function (type) {
    FILE_TYPE_MAP[type].map(function (name) {
      if (name === ext) {
        fileType = type;
      }
    });
  });
  return kebabCase(fileType) + '.svg';
};
