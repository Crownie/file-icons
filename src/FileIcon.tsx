import React, {FunctionComponent} from 'react';
import {getIconType, isMimeType} from './utils';
import {FILE_ICON_MAP} from './constant';
import kebabCase from 'lodash.kebabcase';
import mimeTypes from 'mime-types';

interface Props {
  type: string; // mimetype or file extension e.g.image/jpeg or jpeg
  className?: string;
  style?: any;
}

export const FileIcon: FunctionComponent<Props> = React.memo(
  ({type, className, style = {}}) => {
    const ext = isMimeType(type) ? mimeTypes.extension(type) : type;
    let fileKey = getIconType(ext);
    const fileName = kebabCase(fileKey) + '.svg';
    const fileImg = FILE_ICON_MAP[fileKey];
    return (
      <img
        alt={fileName}
        src={fileImg.default}
        className={className}
        style={style}
      />
    );
  },
);
