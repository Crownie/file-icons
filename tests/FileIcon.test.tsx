import React from 'react';
import '@testing-library/jest-dom';
import {render} from '@testing-library/react';
import {FileIcon} from '../src';

const testTypes = (type, fileName) => {
  let res = render(<FileIcon type={type} />);
  let container = res.container;
  let img = container.getElementsByTagName('img');
  expect(img).toBeDefined();
  expect(img[0].getAttribute('alt')).toEqual(fileName);
};

it('renders correctly', () => {
  let res = render(
    <FileIcon
      type={'pdf'}
      className={'dummy-class'}
      style={{width: 20, height: 20}}
    />,
  );
  let container = res.container;
  expect(container.innerHTML).toMatchSnapshot();
});

it('test file types', () => {
  testTypes('jpeg', 'img.svg');
  testTypes('image/jpeg', 'img.svg');
  testTypes('image/png', 'img.svg');
  testTypes('mp3', 'music.svg');
  testTypes('audio/mp3', 'music.svg');
  testTypes('audio/wav', 'music.svg');
  testTypes('audio/unknown', 'common.svg');
  testTypes('common', 'common.svg');
});
