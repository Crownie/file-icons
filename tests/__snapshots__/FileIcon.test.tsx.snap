// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `"<img alt=\\"pdf.svg\\" src=\\"[object Object]\\" class=\\"dummy-class\\" style=\\"width: 20px; height: 20px;\\">"`;
